libp2p/src/main/kotlin/io/libp2p/multistream/MultistreamImpl.kt

```diff
@@ -6,7 +6,6 @@ import io.libp2p.core.multistream.Multistream
 import io.libp2p.core.multistream.ProtocolBinding
 import java.time.Duration
 import java.util.concurrent.CompletableFuture
-import java.util.concurrent.CopyOnWriteArrayList
 
 class MultistreamImpl<TController>(
```

Comment:
For simplicity can just add the `bindings` to the constructor
```kotlin
class MultistreamImpl<TController>(
    override val bindings: MutableList<ProtocolBinding<TController>>,
    val preHandler: P2PChannelHandler<*>? = null,
    val postHandler: P2PChannelHandler<*>? = null,
    val negotiationTimeLimit: Duration = DEFAULT_NEGOTIATION_TIME_LIMIT
)
```
and remove
`override val bindings: List<ProtocolBinding<TController>> = initList`

Reply:
Agree, it's not that right to pass a `List` (which is implicitly immutable) which is mutated behind the scene and is expected to have effect

Reply:
Sure, I can make that change, though it ripples through the classes a bit.

Reply:
Actually I tried to do this and it goes down a long rabbit hole that I couldn't get to work.

Reply:
Actually I think I have already come up with some more or less acceptable change. I'll try to polish it tomorrow

Reply:
Well, I would suggest to merge that variant with a couple of updates: https://github.com/Peergos/jvm-libp2p/pull/13